### PR TITLE
Ciudades abiertas/smtp configuration section

### PR DESCRIPTION
**Pull request by taitus**
## References
Related Issues: #21 
Related Installer PRs: [Update email configuration #109](https://github.com/consul/installer/pull/109)

## Objectives
Move SMTP configuration from config/environments files to Settings
- config/environments/production.rb 
- config/environments/preproduction.rb
- config/environments/staging.rb

In this PR the application will change from loading the SMTP configuration of the configuration files(`config/environments/xxx.rb`) to loading it from the new Settings page(SMTP configuration). 

Previously this configuration was loaded when the application was starting. Now this configuration is loaded into the application before each email is sent in order to use the values saved in the database from the new SMTP configuration page.

## Visual Changes
Captura de pantalla 2019-09-03 a las 13 08 53
Captura de pantalla 2019-09-19 a las 14 43 31

## Notes
Only for existing installations❗️❗️❗️
**Case 1:**
For existing installations that have the email smtp configuration correctly updated in the symbolic link(for example `shared/config/environments/production.rb`) **it will be necessary to execute** the following `rake task` for the sending of emails to continue working correctly. 
This rake task copies the SMTP configuration of your server to the new Settings that the application needs in order to load the email configuration correctly.
Please execute next command `rake settings:update_smtp_settings`
Remember that until you execute the rake task the sending of emails will be disabled.
You can validate that the rake task has been executed correctly by accessing the administration panel in the `Settings > SMTP configuration` section and seeing that all the fields are correctly filled in.

**Case 2:**
If your existing installation use default ActionMailer::Base.delivery_method and default ActionMailer::Base.smtp_settings and you manage internally the sending of emails
with redirections or another stuff, **you must not execute the rake task**. 
When this feature is disabled the sending of emails keep behaving in the same way.

### CODE REVIEW

File: lib/tasks/settings.rake [hunk 1]
```diff
@@ -67,4 +67,18 @@ namespace :settings do
     Setting["advanced.auth.http_basic_password"] = Rails.application.secrets["http_basic_password"]
   end
 
+  desc "Retrocompatibility smtp settings for existing installations"
```
Comment (Senen) [inline lib/tasks/settings.rake]:
```suggestion
  desc "Copy existing SMTP configuration to settings database"
```

File: config/locales/en/admin.yml [hunk 2]
```diff
@@ -1341,6 +1341,10 @@ en:
           description: "Allow update advanced configuration"
           auth: "Http basic authentication"
           tracking: "Error tracking"
+        smtp:
+          title: "SMTP Configuration"
+          description: "Allow define SMTP configuration to send emails."
```
Comment (Senen) [inline config/locales/en/admin.yml]:
```suggestion
          description: "Set SMTP server configuration to send emails."
```

File: config/locales/es/admin.yml [hunk 3]
```diff
@@ -1342,6 +1342,10 @@ es:
           description: "Permitir actualizar las configuraciones avanzadas"
           auth: "Autenticación básica por http"
           tracking: "Monitorización de errores"
+        smtp:
+          title: "Configuración SMTP"
+          description: "Permitir actualizar la configuración SMTP para el envio de emails."
```
Comment (Senen) [inline config/locales/es/admin.yml]:
```suggestion
          description: "Configuración del servidor SMTP para el envio de emails."
```

File: config/locales/en/settings.yml [hunk 4]
```diff
@@ -116,6 +116,8 @@ en:
       translation_interface_description: "Displays the manual translation interface that allows users to enter their content in all application available languages. This option only affects user application forms and independently will always be active in the administration panel."
       remote_census: "Configure connection to remote census (SOAP)"
       remote_census_description: "Allows to configure the connection to the remote census of each institution"
+      smtp_configuration: "SMTP Configuration"
+      smtp_configuration_description: "Activate this feature to add SMTP configuration and send emails from the application."
```
Comment (Senen) [inline config/locales/en/settings.yml]:
```suggestion
      smtp_configuration_description: "Enable this feature to send emails through SMTP server."
```

File: config/locales/en/settings.yml [hunk 5]
```diff
@@ -232,3 +234,18 @@ en:
       tracking:
         rollbar_server_token: "Rollbar server token"
         rollbar_server_token_description: "This token will give the application access to your Rollbar account."
+    smtp:
+      address: "SMTP Host"
+      address_description: "Configure SMTP Host. Example: 'smtp.example.com'"
```
Comment (Senen) [inline config/locales/en/settings.yml]:
```suggestion
      address_description: "Configure SMTP server host. Example: 'smtp.example.com'"
```

File: config/locales/en/settings.yml [hunk 6]
```diff
@@ -232,3 +234,18 @@ en:
...
+      port: "SMTP Port"
+      port_description: "Configure SMTP Port. Example: 587"
```
Comment (Senen) [inline config/locales/en/settings.yml]:
```suggestion
      port_description: "Configure SMTP server port. Example: 587"
```

File: config/locales/en/settings.yml [hunk 7]
```diff
@@ -232,3 +234,18 @@ en:
...
+      domain: "Domain"
+      domain_description: "Domain to allow configure SMTP. Example: 'example.com'"
```
Comment (Senen) [inline config/locales/en/settings.yml]:
```suggestion
      domain_description: "Configure SMTP server domain. Example: 'example.com'"
```

File: config/locales/en/settings.yml [hunk 8]
```diff
@@ -232,3 +234,18 @@ en:
...
+      username: "SMTP User"
+      username_description: "Configure SMTP User. Example: 'username'"
```
Comment (Senen) [inline config/locales/en/settings.yml]:
```suggestion
      username_description: "Configure SMTP server user. Example: 'username'"
```

File: config/locales/en/settings.yml [hunk 9]
```diff
@@ -232,3 +234,18 @@ en:
...
+      password: "SMTP Password"
+      password_description: "Configure SMTP Password. Example: 'password'"
```
Comment (Senen) [inline config/locales/en/settings.yml]:
```suggestion
      password_description: "Configure SMTP server password. Example: 'password'"
```

File: config/locales/en/settings.yml [hunk 10]
```diff
@@ -232,3 +234,18 @@ en:
...
+      authentication: "SMTP Authentication"
+      authentication_description: "Configure SMTP Authentication kind. Example: 'plain'"
```
Comment (Senen) [inline config/locales/en/settings.yml]:
```suggestion
      authentication_description: "Configure SMTP server authentication type. Example: 'plain'"
```

File: config/locales/en/settings.yml [hunk 11]
```diff
@@ -232,3 +234,18 @@ en:
...
+      enable_starttls_auto: "Enable SMTP TLS"
+      enable_starttls_auto_description: "Allow Enable TLS SMTP configuration."
```
Comment (Senen) [inline config/locales/en/settings.yml]:
```suggestion
      enable_starttls_auto_description: "Configure SMTP server TLS configuration. Example: 'true'"
```

File: config/locales/es/admin.yml [hunk 12]
```diff
@@ -1342,6 +1342,10 @@ es:
...
+          how_to_enable: 'Para actualizar la configuración SMTP se debe activar "Configurar SMTP" en la pestaña "Funcionalidades".'
```
Comment (Senen) [inline config/locales/es/admin.yml]:
```suggestion
          how_to_enable: 'Para activar la configuración SMTP debe activar "Configurar SMTP" en la pestaña "Funcionalidades".'
```

File: config/locales/es/settings.yml [hunk 13]
```diff
@@ -232,3 +234,18 @@ es:
       tracking:
         rollbar_server_token: "Token de acceso a Rollbar"
         rollbar_server_token_description: "Este token le dará a la aplicación acceso a su cuenta Rollbar."
+    smtp:
+      address: "Host SMTP"
+      address_description: "Host para la configuración SMTP. Ejemplo: 'smtp.example.com'"
```
Comment (Senen) [inline config/locales/es/settings.yml]:
```suggestion
      address_description: "Configura el host del servidor SMTP. Ejemplo: 'smtp.example.com'"
```

File: config/locales/es/settings.yml [hunk 14]
```diff
@@ -232,3 +234,18 @@ es:
...
+      port: "Puerto SMTP"
+      port_description: "Puerto para la configuración SMTP. Ejemplo: 587"
```
Comment (Senen) [inline config/locales/es/settings.yml]:
```suggestion
      port_description: "Configura el puerto del servidor SMTP. Ejemplo: 587"
```

File: config/locales/es/settings.yml [hunk 15]
```diff
@@ -232,3 +234,18 @@ es:
...
+      domain: "Dominio"
+      domain_description: "Dominio para la configuración SMTP. Ejemplo: 'example.com'"
```
Comment (Senen) [inline config/locales/es/settings.yml]:
```suggestion
      domain_description: "Configura el dominio del servidor SMTP. Ejemplo: 'example.com'"
```

File: config/locales/es/settings.yml [hunk 16]
```diff
@@ -232,3 +234,18 @@ es:
...
+      username: "Usuario SMTP"
+      username_description: "Usuario para la configuración SMTP. Ejemplo: 'username'"
```
Comment (Senen) [inline config/locales/es/settings.yml]:
```suggestion
      username_description: "Configura el usuario del servidor SMTP. Ejemplo: 'username'"
```

File: config/locales/es/settings.yml [hunk 17]
```diff
@@ -232,3 +234,18 @@ es:
...
+      password: "Contraseña SMTP"
+      password_description: "Contraseña para la configuración SMTP. Ejemplo: 'password'"
```
Comment (Senen) [inline config/locales/es/settings.yml]:
```suggestion
      password_description: "Configura la contraseña del servidor SMTP. Ejemplo: 'password'"
```

File: config/locales/es/settings.yml [hunk 18]
```diff
@@ -232,3 +234,18 @@ es:
...
+      authentication: "Autenticación SMTP"
+      authentication_description: "Tipo de autenticación para la configuración SMTP. Ejemplo: 'plain'"
```
Comment (Senen) [inline config/locales/es/settings.yml]:
```suggestion
      authentication_description: "Configura el mecanismo de autenticación del servidor SMTP. Ejemplo: 'plain'"
```

File: config/locales/es/settings.yml [hunk 19]
```diff
@@ -232,3 +234,18 @@ es:
...
+      enable_starttls_auto: "Activar TLS para SMTP"
+      enable_starttls_auto_description: "Activar TLS para la configuración SMTP."
```
Comment (Senen) [inline config/locales/es/settings.yml]:
```suggestion
      enable_starttls_auto_description: "Activar TLS del servidor SMTP."
```

File: spec/features/admin/settings_spec.rb [hunk 20]
```diff
@@ -47,6 +47,10 @@
     expect(page).to have_content "Advanced Configuration"
     expect(page).to have_content "Allow update advanced configuration"
     expect(page).to have_link("Configure", href: admin_setting_path("advanced"))
+
+    expect(page).to have_content "SMTP Configuration"
+    expect(page).to have_content "Allow define SMTP configuration to send emails."
```
Comment (taitus) [inline spec/features/admin/settings_spec.rb]:
```suggestion
    expect(page).to have_content "Set SMTP server configuration to send emails."
```

**Review by Senen — CHANGES_REQUESTED** [review]
Hi @taitus i added some change suggestions, all of them are about translations.

**Pull request merged**